internal/multigitter/run.go

```diff
@@ -178,9 +183,16 @@ func (r *Runner) runSingleRepo(ctx context.Context, repo domain.Repository) (dom
 		baseBranch = repo.DefaultBranch()
 	}
 
-	err = sourceController.Clone(repo.URL(r.Token), baseBranch)
-	if err != nil {
-		return nil, err
+	if strings.TrimSpace(r.Username) == "" {
+		err = sourceController.Clone(repo.URL(r.Token), repo.DefaultBranch())
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		err = sourceController.Clone(repo.URLWithUsername(r.Username, r.Token), repo.DefaultBranch())
```

Comment:
I like the first option I think. I will try that.

Reply:
The first option seemed harder when trying to implement. I followed the second options but had to rename `URL` to `CloneURL` otherwise the following lines always matched the `URL` interface and broke the tests.

1. https://github.com/lindell/multi-gitter/blob/fcc012fa66187d0968e252d8068fa981a7f72241/internal/multigitter/repocounter/counter.go#L69
2. https://github.com/lindell/multi-gitter/blob/fcc012fa66187d0968e252d8068fa981a7f72241/internal/multigitter/status.go#L28